### PR TITLE
Preventing multiple params in tree

### PR DESCRIPTION
**Pull request by tfwnicholson**
This addresses the issue of multiple parameters being in the tree (#349) which was causing ```longname``` to crash. It was decided with @jameshensman that rather than delaying until the construction of the name to detect and maybe recover from multiple parameters in the tree, that we should detect and abort as soon as someone tries to add the same param to the same tree.

This PR therefore adds the invariant to a tree that a ```Param```/ ```Parameterized``` may only appear that a single point.

Cheers.

### CODE REVIEW

File: GPflow/model.py [hunk 1]
```diff
@@ -348,9 +348,9 @@ class GPModel(Model):
     """
 
     def __init__(self, X, Y, kern, likelihood, mean_function, name='model'):
+        Model.__init__(self, name)
         self.kern, self.likelihood, self.mean_function = \
             kern, likelihood, mean_function
-        Model.__init__(self, name)
```
Comment (jameshensman) [inline GPflow/model.py]: Good catch

File: testing/test_param.py [hunk 2]
```diff
@@ -318,6 +311,120 @@ def testTFMode(self):
...
+
+    def testAddingToAnother(self):
+        """
+        Adding the same Paramterized object to another tree is fine.
+        """
+        m1 = GPflow.param.Parameterized()
+        m1.foo = GPflow.param.Parameterized()
+
+        m2 = GPflow.param.Parameterized()
+        m2.foo = GPflow.param.Parameterized()
```
Comment (jameshensman) [inline testing/test_param.py]: Did you mean `m2.foo = m1.foo` ?

File: testing/test_param.py [hunk 3]
```diff
@@ -318,6 +311,120 @@ def testTFMode(self):
...
+
+    def testAddingToAnother(self):
+        """
+        Adding the same Param object to another tree is fine.
+        """
+        m1 = GPflow.param.Parameterized()
+        m1.foo = GPflow.param.Param(1)
+
+        m2 = GPflow.param.Parameterized()
+        m2.foo = GPflow.param.Param(1)
```
Comment (jameshensman) [inline testing/test_param.py]: again, I think you meant `m2.foo = m1.foo`.

**Comment by jameshensman** [comment]
The main bit of code looks good. I have added a few queries for the tests though.

**Comment by jameshensman** [comment]
looks good to me, @alexggmatthews please have a look at merging this. Thanks.

**Comment by alexggmatthews** [comment]
Hi all, happy to look into this. It will have to be Monday morning now though I'm afraid. Have a good weekend,
Alex

**Comment by alexggmatthews** [comment]
I've moved this to #370 to allow the Continuous Integration to run.

**Comment by alexggmatthews** [comment]
Merged in #370

**Pull request closed without merging**